### PR TITLE
update changelog for 4.15.2

### PR DESCRIPTION
**Pull request by danrosenthal**

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -14,7 +14,4 @@
 
 ### Code quality
 
-- Removed various type assertions and bumped test coverage ([#2638](https://github.com/Shopify/polaris-react/pull/2638))
```
Comment (amrocha) [inline UNRELEASED.md]: Did you miss this one? Or did you cherry pick it out?

Comment (danrosenthal) [reply UNRELEASED.md]: It was mistakenly added back in https://github.com/Shopify/polaris-react/pull/2800

Comment (danrosenthal) [reply UNRELEASED.md]: https://github.com/Shopify/polaris-react/pull/2819/files#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR49

Comment (danrosenthal) [reply UNRELEASED.md]: Here it is: https://github.com/Shopify/polaris-react/blob/master/CHANGELOG.md#code-quality

**Pull request merged**